### PR TITLE
feat: display message if no search results were found in symbol-autocomplete

### LINKED ISSUE #3779: [Feature] Show message if no results have been found in activity dialog
### Steps to reproduce

* Go to _Portfolio_ > _Activities_
* Click the + button
* Search for `asdlkfj` in the _Name, symbol or ISIN_ input field

Add a synthetic (disabled) option (`mat-option`) to the [symbol autocomplete component](https://github.com/ghostfolio/ghostfolio/tree/main/libs/ui/src/lib/symbol-autocomplete) if no results have been found with the following label:

> Oops! Could not find any assets.

### PR DESCRIPTION
**Pull request by dw-0**
Fixes #3779 

## Before:
image

## After:
24-10-01_19-40-38_brave

Signed-off-by: Dominik Willner <th33xitus@gmail.com>

### CODE REVIEW

File: apps/client/src/locales/messages.de.xlf [hunk 1]
```diff
@@ -7258,6 +7258,14 @@
           <context context-type="linenumber">280</context>
         </context-group>
       </trans-unit>
+      <trans-unit id="b225488f8b209e9704760dc9f5d99845a5d07bf6" datatype="html">
+        <source>Oops! Could not find any assets.</source>
+        <target state="translated">Hoppla! Es konnten keine Assets gefunden werden.</target>
```
Comment (dtslvr) [inline apps/client/src/locales/messages.de.xlf:7263]: In the whole app we are using _Ups!_, can you change for consistency reasons? 😃

File: libs/ui/src/lib/symbol-autocomplete/symbol-autocomplete.component.html [hunk 2]
```diff
@@ -12,6 +12,11 @@
   (optionSelected)="onUpdateSymbol($event)"
 >
   @if (!isLoading) {
+    @if (!empty && !filteredLookupItems.length) {
```
Comment (dtslvr) [inline libs/ui/src/lib/symbol-autocomplete/symbol-autocomplete.component.html:15]: Technically this works perfectly. However, I would not show the message for a single character. As the search only provides results for a [query length > 1](https://github.com/ghostfolio/ghostfolio/blob/main/libs/ui/src/lib/symbol-autocomplete/symbol-autocomplete.component.ts#L123), it is a bit annoying.

**Review by dtslvr — CHANGES_REQUESTED** [review]
Thank you for the contribution, @dw-0! It already works, while testing I have found potential improvements.

File: apps/client/src/locales/messages.de.xlf (hunk 1 above)
Comment (dw-0) [reply apps/client/src/locales/messages.de.xlf]: Oh, yeah of course! Sorry for that oversight.

File: libs/ui/src/lib/symbol-autocomplete/symbol-autocomplete.component.html (hunk 2 above)
Comment (dw-0) [reply libs/ui/src/lib/symbol-autocomplete/symbol-autocomplete.component.html]: You are right.
So for fixing this, i introduced a new variable. The reason behind that is, that checking for the input being larger than 1 character results in the message popping up for the debounce time of 400ms, that looked wrong to me. So now a flag is set after the loading is done, inside the subscribe, which then checks for the result length. That way it works quite nicely.

**Review by dtslvr — APPROVED** [review]
I have refactored the code to get rid of the additional variable.

Ready to merge now.

**Comment by dw-0** [comment]
Oh, i haven't had @empty on my radar while writing the code. That's actually quite nice. Will keep that in mind for the future :)

**Comment by dtslvr** [comment]
Thanks again for your contribution, @dw-0! 🖤

**Pull request merged**